### DISCUSSION TITLE
[Device Support Request] TS0601_TZE204_x8fp01wi / Home Assistant  Tuya WiFi ZigBee Three Phase Two-way Bidirectional Energy Meter

### DISCUSSION THREAD

**Discussion by ernsm** [zigpy/zha-device-handlers#2925] [Q&A]
I bought from Aliexpress this device from Tuya also available as a white label by the name of EARU. (see link here: https://www.aliexpress.com/item/1005005993765047.html?spm=a2g0o.order_list.order_list_main.17.62fb5c5fluNBQk&gatewayAdapt=glo2deu)

Device Information: TS0601 von _TZE204_x8fp01wi

Although it supports zigbee it does not integrate well in Home Assistant. See the image below

![Screenshot 2024-01-20 110550](https://github.com/zigpy/zha-device-handlers/assets/29502929/8f5a8387-43e9-4009-9c37-8c75d137a168)

I only get LQI and RSSI entities which are not helpful when monitoring energy consumption. I was researching in a couple of forums but with lesser success. There are similar requests/ideas but for other not yet supported Tuya Energy Meters. 

See conversations here:
https://community.home-assistant.io/t/i-have-a-tuya-smart-zigbee-energy-meter-connected-to-home-assistant-using-a-sonoff-zigbee-gateway-however-it-only-displays-diagnostic-information-such-as-lqi-and-rssi-the-sensor-information-is-not-visible/631561

https://github.com/zigpy/zha-device-handlers/issues/1973

But I don't get the custom quirk loaded based on the guidelines I have found. Another solution might be to buy tuya zigbee router and try using the Tuya HA custom integration.

So I hope someone here is able to help me because I feel that my capability to solve this are very limited.

Thanks in Advance.

**Comment by Hedda** [comment]
> There are similar requests/ideas but for other not yet supported Tuya Energy Meters.

If there is not one already then submit a new a new "Device support request" under issues:

https://github.com/zigpy/zha-device-handlers/issues -> https://github.com/zigpy/zha-device-handlers/issues/new/choose

Why detailed information is needed is explained in the ZHA integration document on the Home Assistant website:

https://www.home-assistant.io/integrations/zha#how-to-add-support-for-new-and-unsupported-devices

and more background information here on why some Zigbee devices do not need quirks:

https://www.home-assistant.io/integrations/zha#knowing-which-devices-are-supported
